File: moseoh/_1주차/_5이진변환반복하기/java/Solution.java
package _1주차._5이진변환반복하기.java;

import java.util.Arrays;

public class Solution {
    public static void main(String[] args) {
        String s = "110010101001";
        System.out.println(Arrays.toString(solution(s)));
    }

    public static int[] solution(String s) {
        int count = 0;
        int sum = 0;
        while (!s.equals("1")) {
            sum += count(s, '0');
            count++;
            s = Integer.toBinaryString(s.replaceAll("0", "").length());
        }

        return new int[]{count, sum};
    }

    public static int count(String s, char c) {
        int count = 0;
        for (char ch : s.toCharArray()) {
            if (ch == c) count++;
        }
        return count;
    }
}

File: moseoh/_1주차/_5이진변환반복하기/kotlin/Solution.kt
package _1주차._5이진변환반복하기.kotlin

fun main(args: Array<String>) {
    val s = "110010101001"
    println(Solution().solution(s).contentToString())
}

class Solution {
    fun solution(s: String): IntArray {
        var str = s
        var count = 0
        var sum = 0
        while (str != "1") {
            count++
            sum += str.count { it == '0' }
            str = Integer.toBinaryString(str.replace("0".toRegex(), "").length)
        }

        return intArrayOf(count, sum)
    }
}